src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -30,6 +31,8 @@ internal class SslStreamInternal
         private AsyncProtocolRequest _readProtocolRequest; // cached, reusable AsyncProtocolRequest used for read operations
         private AsyncProtocolRequest _writeProtocolRequest; // cached, reusable AsyncProtocolRequest used for write operations
 
+        private SemaphoreSlim _asyncWriteActiveSemaphore;
```

Comment:
Specifically, the existing code uses _nestedWrite to throw an exception if multiple writes are issued concurrently; in contrast, this change is making it so that it's allowed to issue multiple concurrent write requests, and they just end up getting serialized.  If that's actually a desired change and deviation from the .NET Framework, it should be made as its own separate and deliberate change, and should likely apply to BeginWrite as well.